lib/Documentable/Search.pm6

```diff
@@ -45,10 +56,35 @@ class Documentable::Search {
         @entries
     }
 
+    method calculate-category(@docs, $kind) {
+        if @docs.elems == 1 {
+            return @docs[0].categories[0];
+        } else {
+            return 'Operators' if @docs.map(*.categories).map(* ~~ /'operators'/).any;
+        }
+
+        given $kind {
+            when 'routine' { 'Routines' }
+            when 'syntax'  { 'Syntax'  }
+            default { die $kind }
+        }
+    }
+
+    method consume-entries-by-kind(@entries, $registry, $kind, $category) {
+        @entries.append: $registry.lookup($kind.Str, :by<kind>).map(-> $doc {
+            self.search-entry(:$category, value => $doc.name, url => $doc.url)
+        }).Slip;
+    }
+
     method search-entry(Str :$category, Str :$value, Str :$url is copy) {
+        if %!seen{"$category - $value"}:exists {
+            warn "There is a duplicate index: $category - $value";
+        } else {
+            %!seen{"$category - $value"} = True;
+        }
+
         $url = $.prefix ?? "/" ~ $.prefix ~ $url !! $url;
-        if ($url ~~ /\.$/) { $url = "{$url}.html" }
-        qq[[\{ category: "{$category}", value: "{$value}", url: "{$url}" \}\n]]
+        qq[[\{ category: "{ $category }", value: "{ $value }", url: "{ $url }" \}\n]]
```

Comment:
Why did you eliminate the hack for the pages that end in `.`?
I would really appreciate if, in big PRs such as this one that contain seemingly unrelated changes, some justification for them would be made. This was introduced in issue #125 to solve issue #72; it's there for a reason. I haven't seen anything to indicate that those URLs are no longer generated. So, again, if you think this is really something that should be tackled differently, open an issue and create a separate PR to discuss possible solutions (including, of course, just eliminating it as worthless technical debt that's no longer relevant)

Reply:
Originally I was working on the origin of numerous hacks we have - URLs not being properly escaped && depending on the filename to serve pages, so considered it unnecessary, not anymore.
But given the situation I re-instantiated it until we have something working nicely.